extensions/charts/ranchercisbenchmark.go

```diff
@@ -0,0 +1,215 @@
+	Message string
+}
+
+// ClusterScanReportSpec represents the specification for a cluster scan report.
+type ClusterScanReportSpec struct {
+	ReportJson string
+}
+
+// CisReport is the report structure stored as report json in cluster scan report spec.
+type CisReport struct {
+	Total         int
+	Pass          int
+	Fail          int
+	Skip          int
+	Warn          int
+	NotApplicable int
+	Results       []*Group `json:"results"`
+}
+
+// Group is the result structure stored as report json in Results of CisReport
+type Group struct {
+	ID     string      `yaml:"id" json:"id"`
+	Text   string      `json:"description"`
+	Checks []*CisCheck `json:"checks"`
+}
+
+// CisCheck is the ID, Description and State structure of individual test in cluster scan.
+type CisCheck struct {
+	Id          string
+	Description string
+	State       string
+}
+
+// InstallRancherCisBenchmarkChart is a helper function that installs the rancher-CIS Benchmark chart.
+func InstallRancherCisBenchmarkChart(client *rancher.Client, installOptions *InstallOptions) error {
+	serverSetting, err := client.Management.Setting.ByID(serverURLSettingID)
+	if err != nil {
+		return err
+	}
+
+	registrySetting, err := client.Management.Setting.ByID(defaultRegistrySettingID)
+	if err != nil {
+		return err
+	}
+
+	cisbenchmarkChartInstallActionPayload := &payloadOpts{
+		InstallOptions:  *installOptions,
+		Name:            RancherCisBenchmarkName,
+		Namespace:       RancherCisBenchmarkNamespace,
+		Host:            serverSetting.Value,
+		DefaultRegistry: registrySetting.Value,
+	}
+
+	chartInstallAction := newCisBenchmarkChartInstallAction(cisbenchmarkChartInstallActionPayload)
+
+	catalogClient, err := client.GetClusterCatalogClient(installOptions.ClusterID)
+	if err != nil {
+		return err
+	}
+
+	// Install the Rancher CIS Benchmark chart
+	err = catalogClient.InstallChart(chartInstallAction, catalog.RancherChartRepo)
+	if err != nil {
+		return err
+	}
+
+	// Wait for chart to be fully deployed
+	watchAppInterface, err := catalogClient.Apps(RancherCisBenchmarkNamespace).Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + RancherCisBenchmarkName,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+		app := event.Object.(*catalogv1.App)
+
+		state := app.Status.Summary.State
+		if state == string(catalogv1.StatusDeployed) {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Register cleanup function for uninstallation
+	client.Session.RegisterCleanupFunc(func() error {
+		return UninstallRancherCisBenchmarkChart(client, installOptions)
+	})
+
+	return nil
+}
+
+// UninstallRancherCisBenchmarkChart is a helper function that uninstalls the rancher-CIS Benchmark chart.
+func UninstallRancherCisBenchmarkChart(client *rancher.Client, installOptions *InstallOptions) error {
```

Comment:
is there a reason this is specific to cis chart? it seems like you could enter the chart parameters as variables, and make a generic uninstallChart function that could be used instead of specific functions for each uninstall.